### ISSUE TITLE
Duplicate middleware execution

### ISSUE THREAD

**Issue by dtuyenle** [jeremydaly/lambda-api#108]
History PR Reference
#76 

It seems like all of the middlewares will run twice. If you do `app.use`

```javascript
const mid1 = (req, res, next) => {
  console.log('hahah1');
  next();
};

const mid2 = (req, res, next) => {
  console.log('hahah2');
  next();
};

const mid3 = (req, res, next) => {
  console.log('hahah3');
  next();
};

app.use(mid1);
app.use(mid2);
app.use(mid3);

app.get('/*', async (req,res) => {
  return { status: 'ok' }
});
```

But if you chain it in the route. It runs once.

```javascript
const mid1 = (req, res, next) => {
  console.log('hahah1');
  next();
};

const mid2 = (req, res, next) => {
  console.log('hahah2');
  next();
};

const mid3 = (req, res, next) => {
  console.log('hahah3');
  next();
};

app.get('/*', mid1, mid2, mid3, async (req,res) => {
  return { status: 'ok' }
});
```

**Comment by dtuyenle** [comment]
Just a note this is serverless framework I am using. At first. I thought It was just the initial load but then how come it runs only once when you chain it in the route.

**Comment by dtuyenle** [comment]
Hi @jeremydaly Have you had a chance to look at this issue ? Appreciate

**Comment by jeremydaly** [comment]
Hi @dtuyenle,
Not yet, sorry. I am working on big update to Lambda API, but I will look at this this weekend.
- Jeremy

**Comment by dtuyenle** [comment]
Thanks @jeremydaly  we are using your library and having a work around but it would be nice to have a solution for this.
